### PR TITLE
MM-16366 - Prevent attachment duplication on import

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Performing bulk import with the same attachment file creates duplicates.
This PR checks for existing attachments and validates that the new file is indeed NEW by checking name, size and hash

#### Ticket Link

  Fixes https://mattermost.atlassian.net/browse/MM-16366

### CODE REVIEW

File: app/import_functions_test.go [hunk 1]
```diff
@@ -2885,4 +2886,31 @@ func TestImportDirectPostWithAttachments(t *testing.T) {
 	assert.Equal(t, len(attachments), 1)
 	assert.Contains(t, attachments[0].Path, "noteam")
 	AssertFileIdsInPost(attachments, th, t)
+
```
Comment (jespino) [inline app/import_functions_test.go]: I miss the test here for same filename and same filesize but different md5sum.

File: app/import_functions_test.go [hunk 2]
```diff
@@ -2885,4 +2886,31 @@ func TestImportDirectPostWithAttachments(t *testing.T) {
...
+	// attempt to import again with same data, should NOT add an attachment
```
Comment (jespino) [inline app/import_functions_test.go]: Probably would be nice to separate each test cases with `t.Run` calls, to have a clearer division of what each test case means.

**Comment by reflog** [comment]
@jespino all good points. done.

**Comment by lindalumitchell** [comment]
QA testing can happen post-merge on this one; removing QA Review label.

**Pull request merged**